File: backend/src/repositories/search-history.repository.ts
import { db } from '../db/database';

const insertStmt = db.prepare(`
  INSERT INTO search_history (keyword, driver, result_count, created_at)
  VALUES (@keyword, @driver, @resultCount, @createdAt)
`);

export class SearchHistoryRepository {
  create(input: { keyword: string; driver: string; resultCount: number }): void {
    insertStmt.run({
      ...input,
      createdAt: new Date().toISOString(),
    });
  }
}

File: backend_py/app/repositories/search_history_repository.py
from datetime import UTC, datetime

from .database import Database


class SearchHistoryRepository:
    def __init__(self, db: Database):
        self.db = db

    def create(self, keyword: str, driver: str, result_count: int) -> None:
        self.db.execute(
            "INSERT INTO search_history (keyword, driver, result_count, created_at) VALUES (?, ?, ?, ?)",
            (keyword, driver, result_count, datetime.now(UTC).isoformat()),
        )
